core/src/main/scala/iliad/gl/eglInterpreter.scala

```diff
@@ -0,0 +1,62 @@
+package iliad
+package gl
+
+import scala.reflect.ClassTag
+
+import iliad.kernel.platform.EGL14Library
+import iliad.kernel.Buffer //TODO: move this to a different place

+
+import cats._
+import cats.data._
+
+final class EGLInterpreter[NDisp, NWin, Disp, Cfg : ClassTag, Sfc, Ctx]
+    extends (EGL[NDisp, NWin, Disp, Cfg, Sfc, Ctx, ?] ~> Reader[EGL14Library.Aux[NDisp, NWin, Disp, Cfg, Sfc, Ctx], ?]) {
+
+  def reader[A](f: EGL14Library.Aux[NDisp, NWin, Disp, Cfg, Sfc, Ctx] => A): Reader[EGL14Library.Aux[NDisp, NWin, Disp, Cfg, Sfc, Ctx], A] = Reader(f)
+
+  def apply[A](egl: EGL[NDisp, NWin, Disp, Cfg, Sfc, Ctx, A]):
+      Reader[EGL14Library.Aux[NDisp, NWin, Disp, Cfg, Sfc, Ctx], A] = egl match {
+    case EGLGetError => reader(_.eglGetError)
+    case EGLChooseConfig(dpy, attrs, count) => reader { lib =>
+      val s = Buffer.capacity[Int](1)
+      val cfgs = new Array[Cfg](count)
+      lib.eglChooseConfig(dpy, attrs.toArray, cfgs, count, s)
+      val size = s.get()
+      @scala.annotation.tailrec
+      def go(num: Int, xs: List[Cfg]): List[Cfg] = {
+        if(num > 0) go(num - 1, cfgs(num - 1) :: xs) else xs
+      }
+      (go(size, List.empty).reverse, size)
+    }
+    case EGLQueryString(disp, p) => reader(_.eglQueryString(disp, p.value))

```

Comment:
I take it that we want to get the last `size` elements from the array.  So if the array was `Array(a, b, c)` and size was 2, we'd end up with `List(b, c)`.  Is this the expected behaviour from EGL?  Even if it is, isn't there a utility method in the standard library that does this for us?